src/orquestra/sdk/_base/_api.py

```diff
@@ -663,6 +666,124 @@ def get_tasks(self) -> t.Set[TaskRun]:
         }
 
 
+def _parse_max_age(age: t.Optional[str]) -> t.Optional[timedelta]:
+    """Parse a string specifying an age into a timedelta object.
+    If the string cannot be parsed, an exception is raises.
+
+    Args:
+        age: the string to be parsed.
+
+    Raises:
+        ValueError if the age string cannot be parsed
+
+    Returns:
+        datetime.timedelta: the age specified by the 'age' string, as a timedelta.
+        None: if age is None
+    """
+    if age is None:
+        return None
+    time_params = {}
+
+    # time in format "{days}d{hours}h{minutes}m{seconds}s"
+
+    # match one or more digits "\d+?" followed by a single character from the 'units'
+    # list. Capture the digits in a group labelled 'name'
+    time_capture_group = r"(?P<{name}>\d+?)[{units}]"
+
+    re_string = ""
+    name_units = (
+        ("days", "dD"),
+        ("hours", "hH"),
+        ("minutes", "mM"),
+        ("seconds", "sS"),
+    )
+    for name, units in name_units:
+        # match each potential time unit capture group between 0 and 1 times.
+        re_string += rf"({time_capture_group.format(name=name, units=units)})?"
+
+    if parts := re.compile(re_string).fullmatch(age):
+        for name, param in parts.groupdict().items():
+            if param:
+                time_params[name] = int(param)
+        if len(time_params) > 0:
+            return timedelta(**time_params)
+
+    raise ValueError(
+        'Time strings must be in the format "{days}d{hours}h{minutes}m{seconds}s". '
+        "Accepted units are:\n"
+        "- d or D: days\n"
+        "- h or H: hours\n"
+        "- m or M: minutes\n"
+        "- s or S: seconds\n"
+        "For example:\n"
+        '- "9h32m" = 9 hours and 32 minutes,\n'
+        '- "8H6S" = 8 hours and 6 seconds,\n'
+        '- "10m" = 10 minutes,\n'
+        '- "3D6h8M13s" = 3 days, 6 hours, 8 minutes and 13 seconds.'
+    )
+
+
+def list_workflow_runs(
+    config_name: str,
```

Comment:
Use config objects in addtion to `ConfigName`
Also switch `str` to `ConfigName`.